src/audio/copier/copier.c

```diff
@@ -713,8 +736,14 @@ static void copier_free(struct comp_dev *dev)
 		rfree(cd->hd);
 	}
 
+	if (dev->ipc_config.type == SOF_COMP_DAI) {
+		dai_zephyr_free(cd->dd, cd->endpoint[0]);
+		rfree(cd->endpoint[0]);
+		buffer_free(cd->endpoint_buffer[0]);
+	}
+
 	for (i = 0; i < cd->endpoint_num; i++) {
-		if (dev->ipc_config.type != SOF_COMP_HOST || cd->ipc_gtw) {
+		if (dev->ipc_config.type > SOF_COMP_DAI || cd->ipc_gtw) {
```

Comment:
we can remove "dev->ipc_config.type > SOF_COMP_DAI", let it like  " if (cd->ipc_gtw) "

Reply:
I also want to remove, however, there is an internal CI error, I still don't know the reason, for prepare and trigger, if use driver to call, there is no issue, if use patch with current implementation, there is a issue.

This is the reason I still don't remove DAI part yet, just in case.